src/validations/html/htmllint.js

```diff
@@ -46,7 +46,17 @@ const errorMap = {
 
   E008: () => ({reason: 'doctype'}),
 
-  E009: () => ({reason: 'href-style'}),
+  E009: (error, source) => {
+    const lines = source.split('\n');
+    const hrefValExpr = /href="([^"]*)/u;
```

Comment:
I don’t think this expression will match in all cases where we might see this error—for instance, the following all trigger an `E009` along with some other error (except in the first example):

```html
<a href='tel:1231231234'>hello</a>
<a href=tel:1231231234>hello</a>
<a HREF="tel:1231231234">hello</a>
<a href = "tel:1231231234">hello</a>
```

It should be fairly straightforward to generalize the expression to match those cases, but I would suggest also, in the spirit of regular expressions being poor substitutes for parsers, account for the possibility that the regexp won’t match (in which case I would argue it’s fine to have a false positive for a `tel:` link).

Alternatively, you could try feeding the relevant part of the source into an HTML fragment parser, though that’s probably overkill.